other/TYPESCRIPT_USAGE.md

```diff
@@ -10,22 +10,45 @@ Pull requests to improve them are welcome and appreciated. If you've never contr
 
 - [x] useDisplayNameInClassName
 
-### Dynamic styles
+### glamorousComponentFactory
 
-To use dynamic styles with custom props use generics. Example:
+The typings for
+* creating your own glamorous component factories
+* using built-in glamorous component factories
+are complete.
+
+```
+// Creating your own
+glamorous(Component)<Props>(/* styleArgument */)
+
+// Using built-in
+glamorous.div<Props>(/* styleArgument */)
+```
+
+#### glamorousComponentFactory arguments
+
+By providing the typings for Props and Theme to Glamorous when setting up your component factory they will be typed on the props argument for function arguments automatically.
 
 ```ts
-const MyStyledDiv = glamorous.div<{noPadding?: boolean}>(
+interface Props {
+  noPadding?: boolean,
+  theme: { color: string }
+}
+
+const MyStyledDiv = glamorous.div<Props>(
   {
     margin: 1,
   },
-  props => ({
-    padding: props.noPadding ? 0 : 4,
+  ({noPadding, theme}) => ({
+    padding: noPadding ? 0 : 4,
```

Comment:
Would be nicer (though not necessary) if the code style would be consistent here :). I mean
```ts
({ noPadding, theme }) => ({
```
instead of
```ts
({noPadding, theme}) => ({
```
the same for this
```ts
<ThemeProvider theme={{color: 'red'}}>
```